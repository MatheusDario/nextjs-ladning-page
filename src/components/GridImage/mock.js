export default {
  $background: false,
  title: 'GALERIA',
  description: 'Uma breve descrição',
  grid: [
    {
      altText: 'Test 1',
      srcImg: 'https://source.unsplash.com/random/360x360?r=1',
    },
    {
      altText: 'Test 2',
      srcImg: 'https://source.unsplash.com/random/360x360?r=2',
    },
    {
      altText: 'Test 3',
      srcImg: 'https://source.unsplash.com/random/360x360?r=3',
    },
    {
      altText: 'Test 4',
      srcImg: 'https://source.unsplash.com/random/360x360?r=4',
    },
    {
      altText: 'Test 5',
      srcImg: 'https://source.unsplash.com/random/360x360?r=5',
    },
    {
      altText: 'Test 6',
      srcImg: 'https://source.unsplash.com/random/360x360?r=6',
    },
  ],
};
