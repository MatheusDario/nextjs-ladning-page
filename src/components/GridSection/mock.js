export default {
  $background: false,
  title: 'DESCRIÇÃO DOS PRINCIPAIS TRABALHOS',
  description: 'Uma breve descrição do asunto que está por vim',
  grid: [
    {
      title: 'Title 1',
      description: `Lorem ipsum dolor, sit amet consectetur adipisicing elit.
      Doloribus quibusdam laborum officiis iste
      beatae obcaecati repellat, et veritatis provident culpa iusto id
      sequi corrupti magni repellendus, placeat corporis esse architecto.`,
    },
    {
      title: 'Title 2',
      description: `Lorem ipsum dolor, sit amet consectetur adipisicing elit.
      Doloribus quibusdam laborum officiis iste
      beatae obcaecati repellat, et veritatis provident culpa iusto id
      sequi corrupti magni repellendus, placeat corporis esse architecto.`,
    },
    {
      title: 'Title 3',
      description: `Lorem ipsum dolor, sit amet consectetur adipisicing elit.
      Doloribus quibusdam laborum officiis iste
      beatae obcaecati repellat, et veritatis provident culpa iusto id
      sequi corrupti magni repellendus, placeat corporis esse architecto.`,
    },
  ],
};
