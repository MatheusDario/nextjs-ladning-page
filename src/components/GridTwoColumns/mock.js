export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Illum eos ratione,
  velit corporis id blanditiis voluptatibus
  quam, numquam eaque molestiae voluptas nesciunt aut dolores aperiam veniam ea natus repellat quasi!`,
  srcimg: 'assets/images/javascript.svg',
};
