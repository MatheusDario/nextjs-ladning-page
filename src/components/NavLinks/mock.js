export default [
  {
    children: 'Link 1',
    link: '#target1',
    newtab: false,
  },
  {
    children: 'Link 2',
    link: '#target2',
    newtab: false,
  },
  {
    children: 'Link 3',
    link: '#target3',
    newtab: false,
  },
  {
    children: 'Link 4',
    link: '#target4',
    newtab: false,
  },
  {
    children: 'Link 5',
    link: '#target5',
    newtab: false,
  },
  {
    children: 'Link 6',
    link: '#target6',
    newtab: false,
  },
  {
    children: 'Link 7',
    link: '#target7',
    newtab: false,
  },
  {
    children: 'Link 8',
    link: '#target8',
    newtab: false,
  },
];
