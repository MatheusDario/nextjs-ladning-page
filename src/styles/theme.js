export const theme = {
  colors: {
    white: '#f0f0f2',
    blue: '#023059',
    lightblue: '#065473',
    aquablue: '#08748c',
    green: '#02735e',
    lightgreen: '#6ab04c',
    lightgray: ' #cecece',
    red: '#dc143c',
    black: '#000000',
  },
  spacing: {
    xmall: '.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
  },
  font: {
    sizes: {
      xmall: '.8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
    },
    family: {
      default: "'Lato', sans-serif",
      title: "'Montserrat', sans-serif",
    },
  },
  breakpoints: {
    mobile: '@media (max-width:768px)',
    tests: '(max-width:768px)',
  },
};
