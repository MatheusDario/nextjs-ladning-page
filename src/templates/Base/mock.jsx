import { GridSection } from '../../components/GridSection';
import gridMock from '../../components/GridSection/mock';
import linksMock from '../../components/NavLinks/mock';

export const mockBase = {
  children: (
    <>
      <GridSection {...gridMock} $background />
      <GridSection {...gridMock} />
      <GridSection {...gridMock} $background />
      <GridSection {...gridMock} />
      <GridSection {...gridMock} $background />
      <GridSection {...gridMock} />
    </>
  ),
  links: linksMock,
  logoData: {
    text: 'Logo',
    link: '#',
  },
  footerHtml: '<p>Teste do footer</p>',
};
