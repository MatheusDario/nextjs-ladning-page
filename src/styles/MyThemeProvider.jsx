import { ThemeProvider } from "styled-components";
import { theme } from "./theme";
import React from "react";
import { GlobalStyles } from "./global-styles";

export default function MyThemeProvider({ children }) {
  return (
    <ThemeProvider theme={theme}>
      {children}
      <GlobalStyles />
    </ThemeProvider>
  );
}

// MyThemeProvider.propTypes = {
//   children: P.node.isRequired,
// };
