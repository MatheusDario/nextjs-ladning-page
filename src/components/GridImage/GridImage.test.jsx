import { GridImage } from '.';
import { renderTheme } from '../../styles/render-theme';

import mock from './mock';

describe('<GridImage />', () => {
  it('should render grid with title, description and images', () => {
    const { container } = renderTheme(<GridImage {...mock} />);
    expect(container).toMatchSnapshot();
  });

  it('should render grid images without background', () => {
    const { container } = renderTheme(
      <GridImage $background={false} {...mock} />,
    );
    expect(container).toMatchSnapshot();
  });
});
