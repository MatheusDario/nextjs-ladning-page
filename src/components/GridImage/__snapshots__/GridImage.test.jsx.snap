// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridImage /> should render grid images without background 1`] = `
.c3 {
  color: #023059;
  font-size: 5.6rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 >.c4 {
  margin-bottom: 5.6rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 5.6rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  transform: scale(1.2) rotate(10deg);
}

.c0 {
  background: #f0f0f2;
  color: #000000;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          GALERIA
        </h2>
        <div
          class="c4 c5"
        >
          Uma breve descrição
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Test 1"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=1"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 2"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=2"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 3"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=3"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 4"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=4"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 5"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 6"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=6"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<GridImage /> should render grid with title, description and images 1`] = `
.c3 {
  color: #023059;
  font-size: 5.6rem;
  text-transform: uppercase;
}

.c5 {
  font-size: 2.4rem;
}

.c2 >.c4 {
  margin-bottom: 5.6rem;
}

.c6 {
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 5.6rem;
}

.c7 {
  overflow: hidden;
}

.c8 {
  width: 100%;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  transform: scale(1.2) rotate(10deg);
}

.c0 {
  background: #f0f0f2;
  color: #000000;
  min-height: 100vh;
  display: flex;
  align-items: center;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c3 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c6 {
    grid-template-columns: 1fr;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          GALERIA
        </h2>
        <div
          class="c4 c5"
        >
          Uma breve descrição
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <img
              alt="Test 1"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=1"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 2"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=2"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 3"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=3"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 4"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=4"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 5"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=5"
            />
          </div>
          <div
            class="c7"
          >
            <img
              alt="Test 6"
              class="c8"
              src="https://source.unsplash.com/random/360x360?r=6"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
